include/trainer_info.h

```diff
@@ -19,39 +19,291 @@ typedef struct TrainerInfo {
+
+/**
+ * @brief Get the lower half of the trainer's ID.
+ * 
+ * This constitutes only the visible trainer ID.
+ * 
+ * @param info  The trainer info.
+ * @return The lower half of the trainer's ID.
+ */
+u16 TrainerInfo_ID_LowHalf(const TrainerInfo *info);
+
+/**
+ * @brief Set the trainer's gender.
+ * 
+ * @param info      The trainer info.
+ * @param gender    New trainer gender.
+ */
+void TrainerInfo_SetGender(TrainerInfo *info, int gender);
+
+/**
+ * @brief Get the trainer's gender.
+ * 
+ * @param info  The trainer info.
+ * @return The trainer's gender.
+ */
+u32 TrainerInfo_Gender(const TrainerInfo *info);
+
+/**
+ * @brief Check if the trainer has a certain badge.
+ * 
+ * @param info      The trainer info.
+ * @param badgeNum  Number of the badge to check, 0-indexed.
+ * @return TRUE if the trainer has the badge, FALSE otherwise.
+ */
+BOOL TrainerInfo_HasBadge(const TrainerInfo *info, int badgeNum);
+
+/**
+ * @brief Set the flag denoting possession of a given badge.
+ * 
+ * @param info      The trainer info.
+ * @param badgeNum  Number of the badge to set.
+ */
+void TrainerInfo_SetBadge(TrainerInfo *info, int badgeNum);
+
+/**
+ * @brief Get the number of badges owned by the trainer.
+ * 
+ * @param info  The trainer info.
+ * @return Number of badges acquired.
+ */
+int TrainerInfo_BadgeCount(const TrainerInfo *info);
+
+/**
+ * @brief Get the trainer's current money.
+ * 
+ * @param info  The trainer info.
+ * @return How much money the trainer has.
+ */
+u32 TrainerInfo_Money(const TrainerInfo *info);
+
+/**
+ * @brief Set the trainer's current money.
+ * 
+ * @param info  The trainer info.
+ * @param money How much money the trainer should have.
+ * @return The trainer's new current money.
+ */
+u32 TrainerInfo_SetMoney(TrainerInfo *info, u32 money);
+
+/**
+ * @brief Get the trainer's appearance in the UNion Room.
+ * 
+ * @param info
+ * @return The trainer's Union Room apperance.
+ */
+u8 TrainerInfo_Appearance(const TrainerInfo *info);
+
+/**
+ * @brief Set the trainer's appearance in the Union Room.
+ * 
+ * @param info          The trainer info.
+ * @param appearance    New appearance setting.
+ */
+void TrainerInfo_SetAppearance(TrainerInfo *info, u8 appearance);
+
+/**
+ * @brief Give the trainer a specified amount of money.
+ * 
+ * @param info  The trainer info.
+ * @param money The amount of money to give the trainer.
+ * @return The trainer's new current money.
+ */
+u32 TrainerInfo_GiveMoney(TrainerInfo *info, u32 money);
+
+/**
+ * @brief Take a specified amount of money from the trainer.
+ * 
+ * @param info  The trainer info.
+ * @param money The amount of money to take from the trainer.
+ * @return The trainer's new current money.
+ */
+u32 TrainerInfo_TakeMoney(TrainerInfo *info, u32 money);
+
+/**
+ * @brief Get the trainer's game code.
+ * 
+ * @param info  The trainer info.
+ * @return The trainer's game code.
+ */
+u8 TrainerInfo_GameCode(const TrainerInfo *info);
```

Comment:
`TrainerInfo_GetGameCode`

Reply:
Resolving in favor of any further discussion on the above comment.